src/core/include/iDynTree/Core/VectorDynSize.h

```diff
@@ -80,6 +80,18 @@ namespace iDynTree
          */
         VectorDynSize(const VectorDynSize& vec);
 
+#if !defined(SWIG_VERSION) || SWIG_VERSION >= 0x030000
```

Comment:
Just to understand, why the `SWIG_VERSION >= 0x030000` ?

Reply:
Actually I don't know. I'm not an expert of `SWIG` so I copied  what was done here:
https://github.com/robotology/idyntree/blob/c0afdd969bf5e7629b4c39df54880e0785fc7753/src/core/include/iDynTree/Core/VectorDynSize.h#L113-L124

If I've to tell the truth I don't know why we need this. Probably @diegoferigo can give me an explanation

Reply:
I blamed the code and it seems that those if come from https://github.com/robotology/idyntree/pull/476 . I am not sure if anyone is still testing or interested at all in SWIG 2.x compatibility, but for sure they are not harming anyone.

Reply:
I think all the systems we support have at least swig 3. I'd be in favour of removing these preprocessor statements and bump the required swig version in CMake.

Reply:
> I think all the systems we support have at least swig 3. I'd be in favour of removing these preprocessor statements and bump the required swig version in CMake.

So if you (@traversaro) agree I can add this modification. Let me know if it's better to open another PR (probably yes)

Reply:
In `Span.h` there are a lot of `#ifndef SWIG`, e.g. 
https://github.com/robotology/idyntree/blob/5b2b487b123155dfadccef92358cccfea10e3ef2/src/core/include/iDynTree/Core/Span.h#L362-L371

I don't know if this may cause a problems. As far as I understood if `SWIG` is defined that part of the code is not defined, consequentially we cannot use that constructor in the code. Am I missing something?

Reply:
> I think all the systems we support have at least swig 3. I'd be in favour of removing these preprocessor statements and bump the required swig version in CMake.

Also in 2018 that was the case, but some old users asked me to support SWIG 2 for their particular workflow. I think that also them do not use SWIG 2 anymore, but let's wait for them to appear and confirm they do not use SWIG 2 anymore. 

Completely unrelated mention: @francesco-romano . :D

Reply:
> In Span.h there are a lot of #ifndef SWIG, e.g.

Those are different, as they are active for all possible version of SWIG. 

> As far as I understood if SWIG is defined that part of the code is not defined, consequentially we cannot use that constructor in the code. Am I missing something?

The `SWIG` preprocessor define is only defined by the SWIG compiler used to generate the bindings, so it is perfectly fine to use those constructor in the normal code, as `SWIG` is never defined in the regular compilation calls, even if SWIG generation is enabled at the CMake level.

Reply:
If you can keep the `ifdef` on the SWIG version it would be great.
We currently still use SWIG 2.0

Reply:
Thanks @francesco-romano !